models/aeif_cond_beta_multisynapse.cpp

```diff
@@ -144,84 +141,69 @@ aeif_cond_beta_multisynapse::Parameters_::get( DictionaryDatum& d ) const
   def< double >( d, names::I_e, I_e );
   def< double >( d, names::V_peak, V_peak_ );
   def< double >( d, names::gsl_error_tol, gsl_error_tol );
-  def< int >( d, "n_synapses", num_of_receptors_ );
   def< bool >( d, names::has_connections, has_connections_ );
 }
 
 void
 aeif_cond_beta_multisynapse::Parameters_::set( const DictionaryDatum& d )
 {
-
   updateValue< double >( d, names::V_th, V_th );
   updateValue< double >( d, names::V_peak, V_peak_ );
   updateValue< double >( d, names::t_ref, t_ref_ );
   updateValue< double >( d, names::E_L, E_L );
   updateValue< double >( d, names::V_reset, V_reset_ );
-  updateValue< double >( d, names::E_ex, E_ex );
-  updateValue< double >( d, names::E_in, E_in );
 
   updateValue< double >( d, names::C_m, C_m );
   updateValue< double >( d, names::g_L, g_L );
 
-  std::vector< double > tau_tmp;
-  if ( updateValue< std::vector< double > >( d, names::taus_decay, tau_tmp ) )
-  {
-    if ( tau_tmp.size() < taus_decay.size() && has_connections_ == true )
+  const size_t old_n_receptors = n_receptors();
+  bool Erev_flag =
+    updateValue< std::vector< double > >( d, names::E_rev, E_rev );
+  bool taur_flag =
+    updateValue< std::vector< double > >( d, names::taus_rise, taus_rise );
+  bool taud_flag =
+    updateValue< std::vector< double > >( d, names::taus_decay, taus_decay );
+  if ( Erev_flag || taur_flag || taud_flag )
+  { // receptor arrays have been modified
+    if ( ( E_rev.size() != old_n_receptors
+           || taus_rise.size() != old_n_receptors
+           || taus_decay.size() != old_n_receptors )
+      && ( !Erev_flag || !taur_flag || !taud_flag ) )
     {
       throw BadProperty(
-        "The neuron has connections, therefore the number of ports cannot be "
-        "reduced." );
-    }
-    for ( size_t i = 0; i < tau_tmp.size(); ++i )
-    {
-      if ( tau_tmp[ i ] <= 0 )
-      {
-        throw BadProperty(
-          "All synaptic time constants must be strictly positive" );
-      }
+        "If the number of receptor ports is changed, all three arrays "
+        "E_rev, taus_rise and taus_decay must be provided." );
     }
-    taus_decay = tau_tmp;
-    num_of_receptors_ = taus_decay.size();
-    tau_tmp.clear();
-    if ( taus_rise.size() < taus_decay.size() )
+    if ( ( E_rev.size() != taus_rise.size() )
+      || ( E_rev.size() != taus_decay.size() ) )
     {
-      for ( size_t i = 0; i < taus_decay.size(); ++i )
-      {
-        taus_rise.push_back( taus_decay[ i ] / 10. );
-        // if taus_rise is not defined explicitly or if it has less elements
-        // than taus_decay, it will be set to taus_decay/10
-      }
+      throw BadProperty(
+        "The reversal potential, synaptic rise time and synaptic decay time "
+        "arrays must have the same size." );
```

Comment:
About this point, please have a look at my answer in
https://github.com/nest/nest-simulator/pull/516#discussion_r83532267
I agree that if we simply want to check  that the changes are consistent, the code can be made more compact. However I think that we should also think about what error messages would be more helpful for the user.
I think we should distinguish those two erroneous situations:
1) The user types a command that provides only one or two arrays, and they have a different size from the old one. Then the error message should be something like:
"If the number of receptor ports is changed, all three arrays  E_rev, taus_rise and taus_decay must be provided."
2) The user types a command that provides three arrays, but they have inconsistent sizes
Then the error message should be something like:
"The reversal potential, synaptic rise time and synaptic decay time arrays must have the same size."
If we first agree on which error messages should be printed in different situations, then we can discuss on what is the best way to implement it in the code.